pyof/foundation/network_types.py

```diff
@@ -398,6 +398,11 @@ def pack(self, value=None):
         Returns:
             bytes: Binary representation of this instance.
         """
+        if isinstance(self.options, BinaryData):
+            self.options = self.options.pack()
+        if isinstance(self.data, BinaryData):
+            self.data = self.data.pack()
+
```

Comment:
This is not the best approach because it covers only this local and specific problem.
A broad solution would be rewriting `__len__` for the `BinaryData` class for it to return correctly. After that, these lines can be removed.